ark/utils/io_utils_test.py

```diff
@@ -184,13 +176,48 @@ def test_list_folders():
         pathlib.Path(os.path.join(temp_dir, 'test_badfile.txt')).touch()
 
         # test substrs is None (default)
-        get_all = iou.list_folders(temp_dir)
-        assert get_all.sort() == dirnames.sort()
+        get_all = iou.list_folders(temp_dir, exact_match=False)
+        assert sorted(get_all) == sorted(dirnames)
 
         # test substrs is not list (single string)
-        get_txt = iou.list_folders(temp_dir, substrs='_txt')
-        assert get_txt.sort() == dirnames[0:2].sort()
+        get_txt = iou.list_folders(temp_dir, substrs='_txt', exact_match=False)
+        assert sorted(get_txt) == sorted(dirnames[0:2])
 
         # test substrs is list
-        get_test_and_other = iou.list_folders(temp_dir, substrs=['test_', 'other'])
-        assert get_test_and_other.sort() == dirnames[1:].sort()
+        get_test_and_other = iou.list_folders(
+            temp_dir, substrs=['test_', 'other'], exact_match=False
+        )
+        assert sorted(get_test_and_other) == sorted(dirnames[1:])
+
+    # Tests "Exact Substring Matching", `exact_match` = True
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # set up temp_dir subdirs
+        dirnames = [
+            'tf_txt',
+            'othertf_txt',
+            'test_csv',
+            'test_out',
+        ]
+
+        dirnames.sort()
+        for dirname in dirnames:
+            os.mkdir(os.path.join(temp_dir, dirname))
+
+        # add extra file
```

Comment:
None of these tests actually test the condition we care about: the substring the user specifies is contained within multiple folders, and only the folder that exactly matches the substring, not the one that contains it, is returned when exact_match=True

Reply:
So if we have the following folders: `test_csv`, `test_csv2`, and `test_csv3`. And the user calls

```python

list_folders(temp_dir, substrs='test_csv', exact_match=True)
```

we want it to only return the folder `test_csv`?

Reply:
yup exactly. Similar to the testing logic for `list_files`, which is right above.